### ISSUE TITLE
Duplicate bus labels for bus of aggregated generation

### ISSUE THREAD

**Issue by gplssm** [openego/eDisGo#76]
Aggregated generators produce duplicate bus labels (used in PyPSA). 

```
INFO:root:Grid expansion after generator import.
DEBUG:edisgo:Cumulative generator capacity (existing): 484.5 kW
DEBUG:edisgo:==> MV generators
DEBUG:edisgo:18 generators imported.
DEBUG:edisgo:Capacities of 0 of 0 existing generators updated (0 kW).
DEBUG:edisgo:Generator 21545 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 21899 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 21903 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 21941 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 23742 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 23743 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 23789 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 23804 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 25055 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 25057 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 25058 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 25060 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 25073 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 25074 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:Generator 2446865 has no geom entry, EnergyMap's geom entry will be used.
DEBUG:edisgo:18 of 18 new generators added (10046.4 kW).
DEBUG:edisgo:==> LV generators
DEBUG:edisgo:1091 generators imported.
DEBUG:edisgo:Capacities of 0 of 0 existing generators (single units) updated (0 kW).
DEBUG:edisgo:2 of 2 decommissioned generators (single units) removed (16.4 kW).
DEBUG:edisgo:Capacities of 0 of 0 existing generators (in 0 of 0 aggregated units) updated (0 kW).
DEBUG:edisgo:43 of 43 decommissioned generators in aggregated generators removed (468.0 kW).
DEBUG:edisgo:1091 of 1091 new generators added (0 single units, 0 to existing agg. generators and 1091 units as new aggregated generators) (total: 35406.4 kW).
DEBUG:edisgo:Cumulative generator capacity (updated): 27749.6 kW
DEBUG:edisgo:Cumulative capacity of imported generators validated.
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
DEBUG:edisgo:Geo distance is zero, check objects' positions. Distance is set to 1m
Traceback (most recent call last):
  File "/opt/pycharm-community-2017.1.3/helpers/pydev/pydevd.py", line 1591, in <module>
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/opt/pycharm-community-2017.1.3/helpers/pydev/pydevd.py", line 1018, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/opt/pycharm-community-2017.1.3/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/guido/rli_local/eDisGo/edisgo/examples/example_double_reinforcement_modified.py", line 81, in <module>
    network.analyze()
  File "/home/guido/rli_local/eDisGo/edisgo/grid/network.py", line 260, in analyze
    self.pypsa = pypsa_io.to_pypsa(self, mode)
  File "/home/guido/rli_local/eDisGo/edisgo/tools/pypsa_io.py", line 145, in to_pypsa
    _check_topology(components)
  File "/home/guido/rli_local/eDisGo/edisgo/tools/pypsa_io.py", line 887, in _check_topology
    collections.Counter(comps.index.values).items()
ValueError: There are duplicates in the Bus list: ['Bus_Generator_agg-lv_load_area_centre_168202-']
```

Occurs in grid 155, 200, 153, 167, 195, 199, 193

@nesnoj, @birgits: if you have some spare time and can support, I appreciate! :)

**Comment by nesnoj** [comment]
I could not reproduce in MVGD 155..
I can have another look tonight..

**Comment by gplssm** [comment]
Ohhh, maybe it's just a bug I introduced myself...?! Make sure you use latest data "20171025134031".

**Comment by gplssm** [comment]
In order to check if the bug was introduced by a recent code change of mine, I jumped back to 7770d9712cc69e8f2a49583de6eedc77a22dd045. The issue still exists then....

**Comment by nesnoj** [comment]
I used a self-generated dingo dataset (v0.3.0 branch) and the HV-MV station's coordinates of the current run "20171025134031" **do not match**!
Are you sure u used the v0.3.0 branch to create this dataset??

Yes, with your data I run into the same problem...in this case, it might be caused by using your 0.2.10 version data with the new 0.3.0 geno data.
But nevertheless: I digged into this problem and it is quite likely that we run into this one in at least one of the MVGDs - I will fix it tomorrow.

**Comment by gplssm** [comment]
Damn! Thx to meta data it was easy to find out that I generated data based on DP 0.2.10 :-1: I'll give it a new try...

**Comment by nesnoj** [comment]
Which metadata r u talking about?
The imported pkl (the one I just generated) tells me:
```
{'data_version': 'model_draft', 'mv_grid_districts': [155], 'assumptions': {'cos_phi_load': '0.97', 'load_factor_lv_trans_fc_normal': '1.0', 'load_area_sat_conn_dist_ring_mod': '100', 'load_factor_lv_trans_lc_normal': '1.0', 'load_area_sat_string_load_threshold': '1000', 'load_factor_mv_trans_fc_normal': '1.0', 'load_area_sat_buffer_radius_inc': '1000', 'load_factor_mv_trans_lc_malfunc': '1.0', 'max_lv_branch_line': '290', 'load_factor_lv_cable_fc_normal': '1.0', 'load_factor_mv_trans_lc_normal': '0.6', 'load_area_sat_buffer_radius': '2000', 'load_area_count_per_ring': '20', 'load_in_generation_case': '0', 'max_half_ring_length': '28', 'population_per_apartment': '2.3', 'lv_ria_branch_connection_distance': '30', 'branch_line_length_retail_industrial': '400', 'load_area_sat_string_length_threshold': '2000', 'mv_max_v_level_fc_diff_normal': '0.02', 'conn_diff_tolerance': '0.0001', 'generator_buffer_radius_inc': '1000', 'load_factor_mv_cable_fc_normal': '1.0', 'mv_station_v_level_operation': '1.0', 'load_factor_mv_line_fc_normal': '1.0', 'lv_max_v_level_fc_diff_normal': '0.03', 'operator_diff_round_digits': '3', 'load_density_threshold': '1', 'lv_nominal_voltage': '400', 'mv_max_v_level_lc_diff_malfunc': '0.10', 'load_area_threshold': '1', 'apartment_house_branch_ratio': '1.5', 'mv_max_v_level_lc_diff_normal': '0.05', 'load_area_stat_conn_dist_ring_mod': '300', 'branch_line_length_agricultural': '800', 'load_factor_mv_cable_lc_malfunc': '1.0', 'lv_max_v_level_lc_diff_normal': '0.05', 'load_factor_mv_line_lc_malfunc': '1.0', 'load_area_sat_conn_dist_weight': '1', 'load_factor_mv_cable_lc_normal': '0.6', 'load_factor_mv_line_lc_normal': '0.6', 'load_factor_lv_cable_lc_normal': '1.0', 'branch_detour_factor': '1.3', 'generation_in_load_case': '0', 'load_area_sat_load_threshold': '100', 'cos_phi_gen': '1', 'mv_half_ring_count_max': '8', 'generator_buffer_radius': '2000', 'voltage_per_km_threshold': '15'}, 'version': '557b603\n', 'database_tables': OrderedDict([('mv_grid_districts', 'EgoGridMvGriddistrict'), ('lv_load_areas', 'EgoDemandLoadarea'), ('lv_grid_district', 'EgoGridLvGriddistrict'), ('mv_stations', 'EgoGridHvmvSubstation'), ('lv_stations', 'EgoGridMvlvSubstation'), ('re_generators', 't_ego_supply_res_powerplant_sq_mview'), ('conv_generators', 't_ego_supply_conv_powerplant_sq_mview')]), 'run_id': '20171027100132'}
```

**Comment by nesnoj** [comment]
Ah, indeed, in your pkl data_version it says version=0.2.10 and data_version=0.2.10.
In mine, version=557b603\n and data_version=model_draft.

Obviously, we use different ones ;)

**Comment by gplssm** [comment]
Indeed! The was "versioned" written in the config file. A `git pull` resolved it :man_facepalming:

**Comment by nesnoj** [comment]
:1st_place_medal:

**Comment by nesnoj** [comment]
I'm digging, digging, digging...

**Comment by nesnoj** [comment]
Ok, solved in b4893a4956eaf499586851309c68b488877ea8fa.
The bug doesn't show up in 5 (dp v0.3.0) tested grids, including your version of MVGD 155 (dp v.0.2.10).
But MVGD 155 (dp v.0.2.10) does not converge...

**Comment by gplssm** [comment]
Missing convergence has - hopefully - other reasons...
